src/admin/resolvers/mutations.ts

```diff
@@ -50,36 +51,43 @@ import {
   updateCollectionStorySortOrder as dbUpdateCollectionStorySortOrder,
   updateCollectionStoryImageUrl as dbUpdateCollectionStoryImageUrl,
 } from '../../database/mutations';
-import { S3 } from 'aws-sdk';
 import { uploadImage } from '../../aws/upload';
+import { IContext } from '../context';
 
 /**
  * Executes a mutation, catches exceptions and records to sentry and console
- * @param db
+ * @param context
  * @param data
  * @param callback
  * @param imageEntityType
  */
 export async function executeMutation<T, U>(
-  db: PrismaClient,
+  context: IContext,
   data: T,
   callback: (db: PrismaClient, data: T) => Promise<U>,
   imageEntityType: ImageEntityType = undefined
 ): Promise<U> {
   try {
-    const entity = await callback(db, data);
+    const { db, authenticatedUser } = context;
 
-    // Associate the image with the entity if the image entity type is provided
-    // and a record for the image exists
-    if (imageEntityType) {
-      await associateImageWithEntity(
-        db,
-        entity as U & { id: number; imageUrl: string },
-        imageEntityType
+    if (authenticatedUser.hasFullAccess) {
```

Comment:
We could also flip this and check e.g `! authenticatedUser.hasFullAccess` and throw the error and do an early exit but this is fine too

Reply:
i like this suggestion. i think it makes the code more readable.

Reply:
Done!